packages/flutter/lib/src/widgets/async.dart

```diff
@@ -329,15 +347,27 @@ typedef AsyncWidgetBuilder<T> = Widget Function(BuildContext context, AsyncSnaps
 ///
 /// The stream may produce errors, resulting in snapshots of the form:
 ///
-/// * `new AsyncSnapshot<int>.withError(ConnectionState.active, 'some error')`
+/// * `AsyncSnapshot<int>.withError(ConnectionState.active, 'some error')`
 ///
 /// The data and error fields of snapshots produced are only changed when the
 /// state is `ConnectionState.active`.
 ///
 /// The initial snapshot data can be controlled by specifying [initialData].
 /// This should be used to ensure that the first frame has the expected value,
 /// as the builder will always be called before the stream listener has a chance
-/// to be processed.
+/// to be processed. In cases where callers wish to have no initial data, the
+/// [new StreamBuilder.withoutInitialData] constructor may be used. Doing so
+/// may cause the first frame to have a snapshot that contains no data.
```

Comment:
may? definitely will, no?

Reply:
It's possible the stream will have produced data by the first frame. The docs above specifically call out that since the builder is called at the discretion of the Flutter pipeline, callers are not guaranteed to see every snapshot in rendered frames.